keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,981 @@
+to `PodDisruptionBudgetStrict`, high-priority(except for `system-cluster-critical` and `system-node-critical`) 
+pods will not be able to violate the PDB and preempt these pods during the scheduling process. 
+This may result in high-priority(except for `system-cluster-critical` and `system-node-critical`) pods 
+being unable to schedule while low-priority pods continue to run normally. 
+
+Although the above situations may arise, due to the fact that PriorityClass is created and managed by 
+cluster administrators with no permission for application owners to perform actions, administrators 
+are able to uniformly configure according to the requirements. Additionally, implementation will 
+include the addition of additional logging or event descriptions to clearly inform the user of the 
+reason why preemption did not occur.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In order to address the issue mentioned above, a new field `DisruptionPolicy` will 
+be added to `PriorityClass`. Users will be able to set this field to 
+determine whether or not PDB violation will be allowed during scheduler preemption. The 
+default value for this field will be `PodDisruptionBudgetBestEffort`, which is 
+consistent with current behavior.  Additionally, in order to avoid system stability 
+issues, even if the victim's PDB is not allowed to be violated during the preemption 
+process, the preempter with priority ClassName of `system-cluster-critical` or `system-node-critical` 
+will still be able to violate the victim's PDB during preemption.
+
+```go
+type PriorityClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Value int32
+	GlobalDefault bool
+	Description string
+	PreemptionPolicy *core.PreemptionPolicy
+
+	// DisruptionPolicy specifies how pods with this priority class 
+	// may be disrupted when being preempted by a higher priority pod.
+	// Defaults to PodDisruptionBudgetBestEffort if unset.
+	// +optional
+	DisruptionPolicy *core.DisruptionPolicy
+}
+```
+
+Currently, two simple policies are provided in this version. One is `PodDisruptionBudgetBestEffort` 
+which indicates that the scheduler will try to avoid violating the PDB during preemption, 
+but it cannot be guaranteed. The other is `PodDisruptionBudgetStrict` which indicates that the 
```

Comment:
Switch to `SystemCriticalOnly` as mentioned below.

Reply:
And in all other places too (assuming that `SystemCriticalOnly` and `BestEffort` are names we're going with, which actually makes more sense to me, as it better reflects the reality).

Reply:
Sorry. I updated the design by the suggestion from https://github.com/kubernetes/enhancements/pull/3755#discussion_r1094884915 
Pls take a look again.